packages/wallet/api/src/lib-wallet.js

```diff
@@ -556,6 +556,46 @@ export function makeWallet({
       });
   }
 
+  /**
+   * @param { () => Promise } get - The function whose return value to memoize.
+   */
+  const makeMemoizedGetter = get => {
+    let pk;
+    const doGet = async () => {
+      const data = await get();
+      pk.resolve(data);
+    };
+
+    return () => {
+      if (pk === undefined) {
+        pk = makePromiseKit();
+        doGet();
+      }
+      return pk.promise;
+    };
+  };
+
+  const getAttIssuer = makeMemoizedGetter(async () =>
+    E(agoricNames)?.lookup('issuer', 'Attestation'),
+  );
+
+  const getAttBrand = makeMemoizedGetter(async () => {
+    const issuer = await getAttIssuer();
+    return E(issuer).getBrand();
+  });
+
+  const getRunStakePublicFacet = makeMemoizedGetter(async () => {
+    const instance = await E(agoricNames)?.lookup('instance', 'runStake');
+    return E(zoe).getPublicFacet(instance);
+  });
+
+  const attMakerPK = makePromiseKit();
```

Comment:
Worth having a type annotation here

Reply:
Done

Reply:
This looks like the interim kludge we came up with, not the notifier based solution from Apr 13: https://github.com/Agoric/agoric-sdk/issues/4263#issuecomment-1098477427

On the other hand... did that train leave the station? Did we need to add the `walletNotifier` property in the governance proposal that just passed? @michaelfig ?